doc/coog_admin_tuto.rst

```diff
@@ -0,0 +1,663 @@
+Coog Deployment
+===============
+
+.. image:: images/coog_admin_architecture.png
+
+
+**coog-admin** is a utility which allows handling **Coog** deployments and
+easing its administration. In general, each Linux user gets a **Coog**
+deployment. All **Coog** deployment related data is stored in two directories:
+
+* **~/coog-admin**: contains source files and scripts which allow launching,
+updating and watching docker containers. This directory's source files should
+never be changed.
+
+* **~/coog-data**: contains deployment overloads (specific configuration)and
+all data volumes linked to active containers.
+
+In addition to these two directories:
+
+* **~/coog-log**: contains functional logs (batch execution reports,
+backup procedure reports)
+
+Prerequired setup
+-----------------
+
+- a server running on Linux
+- docker
+- github
+- an email utility
+
+Create user and load source files
+---------------------------------
+
+**Coog** deployment is done via **docker**. 
+
+It is highly advised to run **Coog** in another user than root. First of all,
+install a user with no administration privileges on the machine. Let that user
+be *coog*.
+
+.. code-block:: bash
+
+   sudo adduser coog
+
+Add *coog* in **docker** groups
+
+.. code-block:: bash
+
+    sudo usermod -aG docker coog
+
+Configure **git** for *coog*
+
+.. code-block:: bash
+
+    su - coog
+    git config --global user.email "coog@<project>.local"
+    git config --global user.name coog
+
+Open *coog* *.bashrc* file and add the following lines:
+
+.. code-block:: bash
+
+    export COOG_CODE_DIR=~/coog-admin
+    export COOG_DATA_DIR=~/coog-data
+    export VISUAL=<editor>
+    export EDITOR=<editor>
+
+*<editor>* can either be **vi** or **nano**
+COOG_DATA_DIR contains path to *coog-admin* repository and COOG_DATA_DIR path to
+*coog-data* repository. These paths can be changed anytime in .bashrc file.
+
+Do not forget running
+
+.. code-block:: bash
+
+    source .bashrc 
+
+Or logout and login to make sure *bashrc* is properly loaded.
+
+In *coog* home directory, clone *coog-admin* git repository and initialize
+**coog-admin**:
+
+.. code-block:: bash
+
+    git clone https://github.com/coopengo/coog-admin 
+    cd coog-admin
+    ./init
+
+**coog-admin branch** must match branch you will build images in (for example,
+if you build 2.0 images, checkout in coog-2.0 for **coog-admin**).
+
+.. code-block:: bash
+
+    git checkout coog-<version_number>
+
+Load Coog images to deploy
+--------------------------
+
+There are two possible **Coog** images: **coog** (backend and web client) and
```

Comment:
There are 10 kind of images:
- standard dependancies / tools (postgres, redis, nginx, etc...)
- Vendor images
  - coog image: coog backend and sao client (web page)
  - web image: coog api and coog app (web app)
  - unoconv image: a standalone service to convert documents (based on unoconv)

To pull standard deps, you need just to execute ./pull

For vendor images, they can be
- built from coog-admin (you need access to private repos)
- loaded from image files (ask Coopengo)
- pulled from dockerhub (you need acces to private repos)